### PR TITLE
[*] Reduce code size: efficient hashing to avoid memory expansion

### PR DESCRIPTION
**Pull request by TuDo1403**
### Description
This PR reduce code size by using assembly to perform hashing to avoid new memory allocation which is cheaper in gas cost and smaller code size.
### Contract changes

The table below shows the following info:
- **Logic**: the logic is changed.
- **ABI**: the ABI is changed.
- **Init data**: new storage field is declared and needs initializing.
- **Dependent**: needs to be changed due to changes in other contracts.

| **Contract name** | **Logic** | **ABI** | **Init data** | **Dependent** |
|-------------------|:---------:|:-------:|:-------------:|:-------------:|
| BridgeTracking    |           |         |               |               |
| GovernanceAdmin   |           |         |               |               |
| Maintenance       |           |         |               |               |
| SlashIndicator    |           |         |               |               |
| Staking           |           |         |               |               |
| StakingVesting    |           |         |               |               |
| ValidatorSet      |           |         |               |               |

### Checklist
- [x] I have clearly commented on all the main functions following the [NatSpec Format](https://docs.soliditylang.org/en/v0.8.0/natspec-format.html)
- [x] The box that allows repo maintainers to update this PR is checked
- [x] I tested locally to make sure this feature/fix works

### CODE REVIEW

File: contracts/libraries/BridgeOperatorsBallot.sol [hunk 1]
```diff
@@ -45,14 +45,18 @@ library BridgeOperatorsBallot {
   /**
    * @dev Returns hash of the ballot.
    */
-  function hash(BridgeOperatorSet calldata _ballot) internal pure returns (bytes32) {
+  function hash(BridgeOperatorSet memory _ballot) internal pure returns (bytes32 digest) {
```
Comment (nxqbao) [inline contracts/libraries/BridgeOperatorsBallot.sol]: Please change the first param in the struct methods of all libraries (which contains only one struct) to `self` to adapt to the new code convention. 

```suggestion
  function hash(BridgeOperatorSet memory self) internal pure returns (bytes32 digest) {
```

Comment (TuDo1403) [reply contracts/libraries/BridgeOperatorsBallot.sol]: updated.

File: yarn.lock
Comment (ducthotran2010) [inline yarn.lock:1]: I found no package addition in `package.json` file. Is this file change on purpose?

File: logs/storage.txt
Comment (ducthotran2010) [inline logs/storage.txt:1]: I don't think updating the hash mechanism lead to change storage.

File: logs/storage_layout.txt
Comment (ducthotran2010) [inline logs/storage_layout.txt:1]: Same

**Review by ducthotran2010 — CHANGES_REQUESTED** [review]

File: logs/storage.txt
Comment (TuDo1403) [reply logs/storage.txt:1]: ^same

File: logs/storage_layout.txt
Comment (TuDo1403) [reply logs/storage_layout.txt:1]: deleted.

File: yarn.lock
Comment (TuDo1403) [reply yarn.lock:1]: change becauses of changes in branch reduce-contract-size

**Review by ducthotran2010 — APPROVED** [review]
Logic looks good

**Pull request merged**